presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -33,128 +36,163 @@
 import java.sql.SQLXML;
 import java.sql.Time;
 import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.ArrayList;
 import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.jdbc.PrestoResultSet.DATE_FORMATTER;
+import static com.facebook.presto.jdbc.PrestoResultSet.TIMESTAMP_FORMATTER;
+import static com.facebook.presto.jdbc.PrestoResultSet.TIME_FORMATTER;
+import static com.google.common.io.BaseEncoding.base16;
+import static java.lang.String.format;
 
 public class PrestoPreparedStatement
         extends PrestoStatement
         implements PreparedStatement
 {
+    private final Map<Integer, String> parameters = new HashMap<>();
+
     PrestoPreparedStatement(PrestoConnection connection, String sql)
             throws SQLException
     {
         super(connection);
+        execute("PREPARE statement FROM " + sql);
```

Comment:
should not be statement constant unique?  As I understand with this only one prepared statement can be created.

Reply:
Good catch, this is broken